test/e2e/dnslookup_test.go

```diff
@@ -0,0 +1,299 @@
+)
+
+var (
+	// for now we use this maybe this can be an env var from test
+	wslDistroName  = "host-resolver-e2e-test"
+	wslTarballName = "distro-0.21.tar"
+	wslTarballURL  = "https://github.com/rancher-sandbox/rancher-desktop-wsl-distro/releases/download/v0.21/distro-0.21.tar"
+	testSrvAddr    = "127.0.0.1"
+	dnsPort        = "53"
+)
+
+func TestLookupARecords(t *testing.T) { //nolint:funlen
+	tmpDir := t.TempDir()
+
+	t.Logf("Dowloading %v wsl distro tarball", wslTarballName)
+	tarballPath := filepath.Join(tmpDir, wslTarballName)
+
+	err := downloadWSLTarball(tarballPath, wslTarballURL)
+	require.NoErrorf(t, err, "Failed to download wsl distro tarball %v", wslTarballName)
+
+	t.Logf("Creating %v wsl distro", wslDistroName)
+	installCmd := cmdExec(
+		tmpDir,
+		"wsl",
+		"--import",
+		wslDistroName,
+		".",
+		tarballPath)
+	err = installCmd.Run()
+	require.NoErrorf(t, err, "Failed to install distro %v", wslDistroName)
+
+	defer func() {
+		t.Logf("Deleting %v wsl distro", wslDistroName)
+		_, err = cmdRunWithOutput("wsl", "--unregister", wslDistroName)
+		require.NoErrorf(t, err, "Failed to start distro %v", wslDistroName)
+	}()
+
+	// It takes a long time to start a new distro,
+	// 20 sec is a long time but that's actually how long
+	// it takes to start a distro without any flakeyness
+	timeout := time.Second * 20
+	tryInterval := time.Second * 2
+	err = confirm(func() bool {
+		// this is a way to figure out if the distro is running
+		// there is an issue with wsl --list --running output
+		// the buffer is not very useful for string search since it
+		// returns some sort of unicode.
+		out, err := cmdRunWithOutput("wsl", "--distribution", wslDistroName, "--exec", "/bin/wslpath", ".")
+		if err != nil {
+			return false
+		}
+		// remove all the weirdness from WSL output
+		return strings.TrimSpace(out) == "."
+	}, tryInterval, timeout)
+	require.NoErrorf(t, err, "Failed to check if %v wsl distro is running", wslDistroName)
+
+	dnsInfs, err := helper.GetDNSInterfaces()
+	require.NoError(t, err, "Failed getting DNS addrs associated to interfaces")
+
+	// This is to cache all the exsisting DNS addresses
+	guidToDNSAddr, err := cacheExsitingDNSAddrs(dnsInfs)
+	require.NoError(t, err, "Failed caching exsisting DNS server addresses")
+
+	// restore the system DNS servers to the original state
+	defer restoreSystemDNS(t, dnsInfs, guidToDNSAddr)
+
+	t.Log("Updating network interfaces with test DNS server addr")
+	// Update the dns addrs to test server
+	updateSystemDNS(t, dnsInfs)
+
+	t.Log("Building host-resolver host binary")
+	err = buildBinaries("../../...", "windows", tmpDir)
+	require.NoError(t, err, "Failed building host-resolver.exe")
+
+	t.Log("Building host-resolver peer binary")
+	err = buildBinaries("../../...", "linux", tmpDir)
+	require.NoError(t, err, "Failed building host-resolver")
+
+	aRecords := testdns.LoadRecords("../testdata/test-300.csv")
+
+	tcpHandler := testdns.NewHandler(false)
+	tcpHandler.Arecords = aRecords
```

Comment:
Hmm, maybe we could instead do:
```suggestion
	tcpHandler := &testdns.NewHandler{
		Truncate: false,
		Arecords: aRecords,
	}
```
(Of course, that means exporting `Truncate`; not sure we want that…)